core/src/main/java/com/linecorp/armeria/client/DnsResolverGroupBuilder.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.function.Consumer;
+
+import javax.annotation.Nullable;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.client.retry.Backoff;
+import com.linecorp.armeria.internal.TransportType;
+
+import io.netty.channel.EventLoopGroup;
+import io.netty.resolver.AddressResolver;
+import io.netty.resolver.AddressResolverGroup;
+import io.netty.resolver.HostsFileEntriesResolver;
+import io.netty.resolver.ResolvedAddressTypes;
+import io.netty.resolver.dns.DnsCnameCache;
+import io.netty.resolver.dns.DnsNameResolver;
+import io.netty.resolver.dns.DnsNameResolverBuilder;
+import io.netty.resolver.dns.DnsQueryLifecycleObserverFactory;
+import io.netty.resolver.dns.DnsServerAddressStreamProvider;
+import io.netty.resolver.dns.NoopAuthoritativeDnsServerCache;
+import io.netty.resolver.dns.NoopDnsCache;
+import io.netty.resolver.dns.NoopDnsCnameCache;
+
+/**
+ * Builds an {@link AddressResolverGroup} which builds {@link AddressResolver}s that update DNS caches
+ * automatically. A usual DNS resolver such as {@link DnsNameResolver}, a cache is expired after TTL.
```

Comment:
`Standard {@link DnsNameResolver} will only expire a cache entry after TTL, meaning DNS queries after TTL will always take time to resolve. A refreshing {@link AddressResolver} on the other hand updates the DNS cache automatically when TTL elapses, meaning DNS queries after TTL will retrieve a refreshed result right away.`

Reply:
This is the sentence I wanted to write. 😉 Thanks a lot. 🙇